parlai/core/torch_generator_agent.py

```diff
@@ -1005,6 +1005,19 @@ def _get_context(self, batch, batch_idx):
             ctxt = batch.full_text_vec[batch_idx]
         return ctxt
 
+    def _get_batch_context(self, batch):
+        """
+        Version of TGA._get_context() that operates on full batches for speed.
+        """
+        if self.beam_context_block_ngram <= 0:
+            # We aren't context blocking, return empty tensor of the correct size
+            return torch.LongTensor([[]] * batch.batchsize)
```

Comment:
```suggestion
            return torch.zeros(batch.batchsize, 0)
```

Reply:
Yeah, that's cleaner - just added that, but with `dtype=torch.long`